File: js/shortest_unsorted_continuous_subarray.js
function shortest_unsorted_continuous_subarray(list) {
  let i = 0;
  let upPeak = -Infinity;
  let downPeak = Infinity;

  while (i < list.length) {
    if (list[i] > list[i + 1] && upPeak < list[i]) {
      upPeak = list[i];
    }
    i++;
  }

  i = list.length - 1;
  while (i >= 0) {
    if (list[i] < list[i - 1] && downPeak > list[i]) {
      downPeak = list[i];
    }
    i--;
  }

  let left;
  if (downPeak !== Infinity) {
    for (let i = 0; i < list.length; i++) {
      if (downPeak > list[i]) {
        left = i;
      }
    }
  }

  let right;
  if (upPeak !== -Infinity) {
    for (let i = list.length - 1; i >= 0; i--) {
      if (upPeak < list[i]) {
        right = i;
      }
    }
  }

  return left !== undefined ? Math.abs(right - left - 1) : -1;
}

console.log(shortest_unsorted_continuous_subarray([2, 6, 4, 8, 10, 9, 15]));

File: python/shortestUnSortedContinuousSubArray.py
import math
from typing import List

def shortestUnSortedContinuousSubArray(list: List[int]) -> int:
    left = 0;
    right = 0;

    max = -math.inf;
    min = math.inf;

    idx = 0;

    while (idx < len(list) - 1):
        if (list[idx] > list[idx + 1]):
            idx = idx + 1;
            break;
        idx = idx + 1;

    if (idx == len(list) - 1):
        return 0;

    while (idx < len(list)):
        if (list[idx] < min):
            min = list[idx];
        idx = idx + 1;
    idx = 0;
    while (idx < len(list)):
        if (list[idx] > min):
            left = idx;
            break;
        idx = idx + 1;

    idx = len(list) - 1;
    while (idx > 0):
        if (list[idx] < list[idx - 1]):
            idx = idx - 1;
            break;
        idx = idx - 1;
    while (idx >= 0):
        if (list[idx] > max):
            max = list[idx];
        idx = idx - 1;
    idx = len(list) - 1;
    while (idx >= 0):
        if (list[idx] < max):
            right = idx;
            break;
        idx = idx - 1;

    return right - left + 1;
        


print(shortestUnSortedContinuousSubArray([2,6,4,8,10,9,15]));